### PR TITLE
Introduce ObstacleDetector and make changes to MotionConfiguration

### PR DESCRIPTION
**Pull request by nfranczak**
We introduce:
```
// Pairs a vision service with a camera, informing the service about which camera it may use
message ObstacleDetector {
  common.v1.ResourceName vision_service = 1;
  common.v1.ResourceName camera = 2;
}
```
to `motion.proto`

We also update update `MotionConfiguration` and replace `vision_services` with `ObstacleDetector`
```
message MotionConfiguration {
  // The ObstacleDetectors that will be used for transient obstacle avoidance
  repeated ObstacleDetector obstacle_detectors = 1;
  ...
 ```

### CODE REVIEW

File: proto/viam/service/motion/v1/motion.proto [hunk 1]
```diff
@@ -120,9 +120,15 @@ message MoveOnMapResponse {
...
+// Pairs a vision service with a camera, informing the service about which camera it may use
+message ObstacleDetector {
+  common.v1.ResourceName vision_service = 1;
+  common.v1.ResourceName camera = 2;
+}
+
 message MotionConfiguration {
-  // The name of the vision service(s) that will be used to detect obstacles
-  repeated common.v1.ResourceName vision_services = 1;
+  // Pairs a vision service with a camera, informing the service about which camera it may use
```
Comment (tessavitabile) [inline proto/viam/service/motion/v1/motion.proto:130]: [nit] Probably don't need to repeat this comment. Instead, you could say "The ObstacleDetectors that will be used for transient obstacle avoidance"

**Review by tessavitabile — APPROVED** [review]

File: proto/viam/service/motion/v1/motion.proto (hunk 1 above)
Comment (jeremyrhyde) [reply proto/viam/service/motion/v1/motion.proto:130]: Seconded

**Review by jeremyrhyde — APPROVED** [review]
LGTM, thanks for putting this together Nick and Tess for pushing it through quickly!

**Review by biotinker — APPROVED** [review]
LGTM!

**Review by nicksanford — APPROVED** [review]

**Comment by nfranczak** [comment]
closed and re-opened as (https://github.com/viamrobotics/api/pull/380) so we have clean git commit history

**Pull request closed without merging**